pandas/tests/test_internals.py

```diff
@@ -286,6 +288,36 @@ def test_repr(self):
         pass
 
 
+class TestDatetimeBlock(tm.TestCase):
+    _multiprocess_can_split_ = True
+
+    def test_try_coerce_arg(self):
+        block = create_block('datetime', [0])
+
+        # coerce None
+        none_coerced = block._try_coerce_args(block.values, None)[1]
+        self.assertTrue(pd.Timestamp(none_coerced) is pd.NaT)
+
+        # coerce np.datetime64
```

Comment:
put these all in a loop (np.datetime64, datetime, date) and you can use the same comparison checks to avoid some duplicated code